commit fb30b6bbf28d91667dcb9691dfeefffac4f99b82
Author: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date:   Sun Oct 18 00:24:19 2009 +0200

    Fix pthread_kill(thread, 0)
    
    * sysdeps/hurd/pt-kill.c (pthread_kill): Return immediately after checks
    without calling _hurd_raise_signal if sig is 0.

diff --git a/sysdeps/hurd/pt-kill.c b/sysdeps/hurd/pt-kill.c
index f970e06..d204e3f 100644
--- a/libpthread/sysdeps/hurd/pt-kill.c
+++ b/libpthread/sysdeps/hurd/pt-kill.c
@@ -39,6 +39,9 @@ pthread_kill (pthread_t thread, int sig)
   ss = _hurd_thread_sigstate (pthread->kernel_thread);
   assert (ss);
 
+  if (!sig)
+    return 0;
+
   detail.exc = 0;
   detail.code = sig;
   detail.error = 0;
